src/main/java/seedu/address/model/person/Person.java

```diff
@@ -91,7 +91,10 @@ public boolean isSamePerson(Person otherPerson) {
         }
 
         return otherPerson != null
-            && (otherPerson.getNric().equals(getNric()) || otherPerson.getPhone().equals(getPhone()));
+                && otherPerson.getName().equals(getName())
+                && otherPerson.getNric().equals(getNric())
+                && otherPerson.getPhone().equals(getPhone())
+                && otherPerson.getEmail().equals(getEmail());
```

Comment:
There might be an issue with this.
Previously it was getName AND (getNric OR getPhone).
Now it is getName AND getNric AND getPhone AND getEmail.

Reply:
noted